src/hypothesis/strategies.py

```diff
@@ -910,6 +912,108 @@ def do_draw(self, data):
     return PermutationStrategy()
 
 
+@defines_strategy
+def datetimes(min_datetime=dt.datetime.min, max_datetime=dt.datetime.max,
+              timezones=None):
+    """A strategy for generating datetimes, which may be timezone-aware.
+
+    This strategy works by drawing a naive datetime between ``min_datetime``
+    and ``max_datetime``, which must both be naive (have no timezone).
+    If ``timezones`` is None, this value is returned.
+
+    If ``timezones`` is not None, it must be a strategy that generates tzinfo
+    objects (or None, which is valid for naive datetimes).  A value drawn
+    from this strategy will be added to a naive datetime, and the resulting
+    tz-aware datetime returned.
+
+    .. note::
+        tz-aware datetimes from this strategy may be ambiguous or non-existent
+        due to daylight savings, leap seconds, timezone and calendar
+        adjustments, etc.  This is intentional, as malformed timestamps are a
+        common source of bugs.  Consider validating all datetime values, or
+        working purely in UTC to avoid such issues.
+
+    :py:func:`hypothesis.extra.timezones` requires the ``pytz`` package, but
+    provides all timezones in the Olsen database.  If you also want to allow
+    naive datetimes, combine strategies like ``none() | timezones()``.
+
+    Alternatively, you can create a list of the timezones you wish to allow
+    (e.g. from the standard library, ``datetutil``, or ``pytz``) and use
+    :py:func:`sampled_from`.  Ensure that simple values such as None or UTC
+    are at the beginning of the list for proper minimisation.
+
+    """
+    from hypothesis.searchstrategy.datetime import DatetimeStrategy
+
+    check_type(dt.datetime, min_datetime, 'min_datetime')
+    check_type(dt.datetime, max_datetime, 'max_datetime')
+    if min_datetime.tzinfo is not None:
+        raise InvalidArgument('min_datetime=%r must not have tzinfo'
+                              % min_datetime)
```

Comment:
My preference is fairly strongly to always wrap % arguments in a tuple even when it's valid not to FWIW. I know it can't go wrong here, but it's better to be obviously correct than non-obviously correct.